ls1intum/artemis-android
# `Communication`: Allow users to view and download sent attachments

### Problem Description

As described in #117 all non-image attachments that are sent in the chat are rendered as links and cannot be opened currently due to an authorization error, since there is no authToken attached to the request.

### Changes
 

This PR introduces a Link resolver, that is capable of detecting the target of the link and attaches an authToken if needed. If the link targets a pdf file, the resolver passes the request to a new Pdf View, which is wrapped in a modal bottom sheet. Requests that target other Artemis pages should also be opened in a modal bottom sheet in a web view, which allows attaching cookies, however there were issues (see #245). All other links are now opened as new IntentTab.
The new Pdf view was inspired by [this repository](https://github.com/GRizzi91/bouquet). It allows users to share and download the pdf. Pdf can also be shown vertically and horizontally.

Note: The security checks have failed because of [the repository](https://github.com/GRizzi91/bouquet) being temporarily added to the project. However, it has been removed and the pdf viewer was added manually including various modifications (see *pdf* package in `core:ui` module)

### Steps for testing

1. Go into a chat that contains a resource pdf.
2. Click on it and verify, that the pdf view opens the pdf.
3. Try downloading, sharing and rotating.
4. Turn of your internet connection and try loading the pdf again, verify that an appropriate error message is shown.

### Screenshots
![PdfViewer](https://github.com/user-attachments/assets/7737c45c-11da-45ca-9b37-ce815d367f4e)
